core/src/main/scala/kafka/zk/migration/ZkTopicMigrationClient.scala

```diff
@@ -47,8 +47,14 @@ class ZkTopicMigrationClient(zkClient: KafkaZkClient) extends TopicMigrationClie
     if (!interests.contains(TopicVisitorInterest.TOPICS)) {
       throw new IllegalArgumentException("Must specify at least TOPICS in topic visitor interests.")
     }
-    val topics = zkClient.getAllTopicsInCluster()
-    val replicaAssignmentAndTopicIds = zkClient.getReplicaAssignmentAndTopicIdForTopics(topics)
+    val allTopics = zkClient.getAllTopicsInCluster()
+    val topicDeletions = readPendingTopicDeletions().asScala
+    val topicsToMigrated = allTopics -- topicDeletions
+    if (topicDeletions.nonEmpty) {
+      warn(s"Found ${topicDeletions.size} pending topic deletions: $topicDeletions. These will be not migrated " +
```

Comment:
I think we should probably limit listing the topics that are being deleted to like the first 10 or whatever. Otherwise we could generate quite a long log4j message.

Reply:
Yea, i wondered about that. What about logging each deletion separately at TRACE ?

Reply:
if it's too much effort, just log the #